@sebastianelsner Build on Github failed with 

```
Option parsing failed: Unknown option --appimage-extract-and-run
make: *** [Makefile:16: bundle_appimage] Error 1
```
My claude tells me that this patch should fix it [0001-Fix-for-failure-on-Github-CI.patch](https://github.com/user-attachments/files/26792232/0001-Fix-for-failure-on-Github-CI.patch). Could not push it to your fork, got permission error.

In general I have a worry that building the ~5+GB Appimage might fail on Github due the size limit, but if it comes to that I'll handle it via manual uploads to https://www.appimagehub.com/ as that seems the place people get their Appimages. 

Please add this patch, so we can test if maybe the Github will be able to build the Appimage

--- a/appimage/build-appimage.sh
+++ b/appimage/build-appimage.sh
@@ -111,10 +111,10 @@ if [ ! -f "$RUNTIME" ]; then
         "https://github.com/AppImage/type2-runtime/releases/download/continuous/runtime-${ARCH}"
 fi
 
-# Use --appimage-extract-and-run when FUSE is unavailable (CI, containers)
+# Use APPIMAGETOOL_EXTRACT_AND_RUN when FUSE is unavailable (CI, containers)
 EXTRA_ARGS=(--runtime-file "$RUNTIME" --no-appstream)
 if [ "${CI:-}" = "true" ] || ! command -v fusermount &>/dev/null; then
-    EXTRA_ARGS+=(--appimage-extract-and-run)
+    export APPIMAGETOOL_EXTRACT_AND_RUN=1
 fi
 
 # Validate AppStream metadata ourselves in offline mode. appimagetool's internal

--- a/buzz/widgets/application.py
+++ b/buzz/widgets/application.py
@@ -3,6 +3,7 @@ import os
 import sys
 import locale
 import platform
+import threading
 import darkdetect
 
 from posthog import Posthog
@@ -85,7 +86,7 @@ class Application(QApplication):
                 "machine": platform.machine(),
                 "version": platform.version(),
             })
-            posthog.shutdown()
+            threading.Thread(target=posthog.shutdown, daemon=True).start()
 
         logging.debug(f"Launching Buzz: {VERSION}, " 
                       f"locale: {locale.getlocale()}, "

--- a/share/metainfo/io.github.chidiwilliams.Buzz.metainfo.xml
+++ b/share/metainfo/io.github.chidiwilliams.Buzz.metainfo.xml
@@ -26,6 +26,8 @@
 
   <url type="bugtracker">https://github.com/chidiwilliams/buzz/issues</url>
   <url type="homepage">https://github.com/chidiwilliams/buzz</url>
+  <url type="faq">https://chidiwilliams.github.io/buzz/docs</url>
+  <url type="vcs-browser">https://github.com/chidiwilliams/buzz</url>
 
   <branding>
     <color type="primary" scheme_preference="light">#f66151</color>

--- a/tests/model_loader_test.py
+++ b/tests/model_loader_test.py
@@ -182,6 +182,26 @@ class TestTranscriptionModel:
         assert model.model_type in list(ModelType)
         assert model.model_type.is_available() is True
 
+    def test_default_respects_model_type_env(self, monkeypatch):
+        monkeypatch.setenv("BUZZ_DEFAULT_MODEL_TYPE", "Faster Whisper")
+        model = TranscriptionModel.default()
+        assert model.model_type == ModelType.FASTER_WHISPER
+
+    def test_default_respects_model_size_env(self, monkeypatch):
+        monkeypatch.setenv("BUZZ_DEFAULT_MODEL_SIZE", "base")
+        model = TranscriptionModel.default()
+        assert model.whisper_model_size == WhisperModelSize.BASE
+
+    def test_default_invalid_model_type_env_falls_back(self, monkeypatch):
+        monkeypatch.setenv("BUZZ_DEFAULT_MODEL_TYPE", "NotAModelType")
+        model = TranscriptionModel.default()
+        assert model.model_type.is_available() is True
+
+    def test_default_invalid_model_size_env_falls_back(self, monkeypatch):
+        monkeypatch.setenv("BUZZ_DEFAULT_MODEL_SIZE", "not-a-size")
+        model = TranscriptionModel.default()
+        assert model.whisper_model_size == WhisperModelSize.TINY
+
     def test_get_local_model_path_openai_api(self):
         model = TranscriptionModel(model_type=ModelType.OPEN_AI_WHISPER_API)
         assert model.get_local_model_path() == ""

--- a/tests/transcriber/whisper_file_transcriber_test.py
+++ b/tests/transcriber/whisper_file_transcriber_test.py
@@ -428,4 +428,21 @@ class TestWhisperFileTranscriber:
         # Assert that file was not created
         assert os.path.isfile(output_file_path) is False
 
+
+class TestTranscribeFasterWhisper:
+    def test_raises_when_model_path_is_empty(self):
+        task = FileTranscriptionTask(
+            model_path="",
+            transcription_options=TranscriptionOptions(
+                model=TranscriptionModel(
+                    model_type=ModelType.FASTER_WHISPER,
+                    whisper_model_size=WhisperModelSize.TINY,
+                )
+            ),
+            file_transcription_options=FileTranscriptionOptions(file_paths=[test_audio_path]),
+            file_path=test_audio_path,
+        )
+        with pytest.raises(FileNotFoundError, match="BUZZ_MODEL_ROOT"):
+            WhisperFileTranscriber.transcribe_faster_whisper(task)
+
         time.sleep(3)

--- a/tests/widgets/file_transcriber_widget_test.py
+++ b/tests/widgets/file_transcriber_widget_test.py
@@ -48,7 +48,8 @@ class TestFileTranscriberWidget:
         mock_triggered = Mock()
         widget.triggered.connect(mock_triggered)
 
-        with patch("buzz.widgets.transcriber.file_transcriber_widget.show_model_download_error_dialog") as mock_err:
+        with patch("buzz.widgets.transcriber.file_transcriber_widget.show_model_download_error_dialog") as mock_err, \
+             patch.object(widget, "save_preferences"):
             widget.on_model_loaded("")
             mock_err.assert_called_once()
             mock_triggered.assert_not_called()
@@ -63,7 +64,8 @@ class TestFileTranscriberWidget:
         mock_triggered = Mock()
         widget.triggered.connect(mock_triggered)
 
-        with patch("buzz.widgets.transcriber.file_transcriber_widget.show_model_download_error_dialog") as mock_err:
+        with patch("buzz.widgets.transcriber.file_transcriber_widget.show_model_download_error_dialog") as mock_err, \
+             patch.object(widget, "save_preferences"):
             widget.on_model_loaded("")
             mock_err.assert_not_called()
             mock_triggered.assert_called_once()

--- a/tests/widgets/file_transcriber_widget_test.py
+++ b/tests/widgets/file_transcriber_widget_test.py
@@ -1,8 +1,10 @@
-from unittest.mock import Mock
+from unittest.mock import Mock, patch
 
 from PyQt6.QtCore import Qt
 from pytestqt.qtbot import QtBot
 
+from buzz.model_loader import ModelType, TranscriptionModel
+from buzz.transcriber.transcriber import TranscriptionOptions
 from buzz.widgets.transcriber.file_transcriber_widget import FileTranscriberWidget
 from tests.audio import test_audio_path
 
@@ -35,3 +37,33 @@ class TestFileTranscriberWidget:
         assert transcription_options.language is None
         assert file_transcription_options.file_paths == [test_audio_path]
         assert len(model_path) > 0
+
+    def test_on_model_loaded_empty_path_shows_error_for_local_model(self, qtbot: QtBot):
+        widget = FileTranscriberWidget(file_paths=[test_audio_path])
+        qtbot.add_widget(widget)
+        widget.transcription_options = TranscriptionOptions(
+            model=TranscriptionModel(model_type=ModelType.FASTER_WHISPER)
+        )
+
+        mock_triggered = Mock()
+        widget.triggered.connect(mock_triggered)
+
+        with patch("buzz.widgets.transcriber.file_transcriber_widget.show_model_download_error_dialog") as mock_err:
+            widget.on_model_loaded("")
+            mock_err.assert_called_once()
+            mock_triggered.assert_not_called()
+
+    def test_on_model_loaded_empty_path_allowed_for_openai_api(self, qtbot: QtBot):
+        widget = FileTranscriberWidget(file_paths=[test_audio_path])
+        qtbot.add_widget(widget)
+        widget.transcription_options = TranscriptionOptions(
+            model=TranscriptionModel(model_type=ModelType.OPEN_AI_WHISPER_API)
+        )
+
+        mock_triggered = Mock()
+        widget.triggered.connect(mock_triggered)
+
+        with patch("buzz.widgets.transcriber.file_transcriber_widget.show_model_download_error_dialog") as mock_err:
+            widget.on_model_loaded("")
+            mock_err.assert_not_called()
+            mock_triggered.assert_called_once()

